architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,390 @@
+|                              |              | /planes/radius/MyCompany/rolesdefinitions            | *              |
+|                              |              | /planes/radius/MyCompany/roleassignments             | *              |
+| Developer                    | developer    | Applications.Core/applications/applications          | *              |
+|                              |              | Applications.Core/applications/containers            | *              |
+|                              |              | Applications.Core/applications/gateways              | *              |
+|                              |              | Applications.Core/applications/secretStores          | *              |
+|                              |              | Applications.Core/applications/volumes               | *              |
+|                              |              | Applications.Datastores/*                            | *              |
+|                              |              | Applications.Messaging/*                             | *              |
+|                              |              | Applications.Dapr/*                                  | *              |
+| Deployer                     | deployer     | Applications.Core/environments                       | deployTo       |
+| Environment Administrator    | env-admin    | Applications.Core/environments                       | *              |
+|                              |              | Applications.Core/secretStores                       | *              |
+|                              |              | System.AWS/credentials                               | *              |
+|                              |              | System.Azure/credentials                             | *              |
+|                              |              | System.Kubernetes/credentials (future functionality) | *              |
+| Resource Type Administrator  | type-admin   | System.Resources/resourceproviders                   | *              |
+| Recipe Administrator         | recipe-admin | Only the recipe property of environments             | *              |
+
+#### User story 1 – Assigning the Radius Administrator role
+
+> [!NOTE]
+>
+> In a new Radius installation, there is no identity provider (IdP) configured. Radius does not have a built-in identity system, therefore the Radius administrator must configure an IdP before Radius will recognize and authorize the user. Roles can be assigned prior to IdP configuration, but without an IdP there is no effect. 
+
+As a platform engineer, I just installed Radius using my Kubernetes cluster credentials. I want to access Radius via my typical user credentials from my IdP.
+
+**User Experience**
+
+```bash
+# Assign the Radius Administrator role to a user from the IdP
+rad role assignment create \
+  --assignee me@my-company.net \
+  --role radius-admin \
+  --scope /planes/radius/*
```

Comment:
Can scopes be other things besides resource groups? If a scope is an individual resource, then does a role still make sense? For example, would we want to assign `radius-admin` to a single resource or resource group?

Reply:
Yes, scopes can be other things besides a resource group. For example, the resource type admin are alllowed the `System.Resources/resourceproviders/MyCompany.App/*` actions on `/planes/radius/MyCompany/ResourceTypes`.